app/gui2/src/components/TextFormattingSelector.vue

```diff
@@ -0,0 +1,121 @@
+<script setup lang="ts">
+import DropdownMenu from '@/components/DropdownMenu.vue'
+import SvgButton from '@/components/SvgButton.vue'
+import SvgIcon from '@/components/SvgIcon.vue'
+import { ref, watch } from 'vue'
+import { TextFormatOptions } from './visualizations/TableVisualization.vue'
+
+const emit = defineEmits<{
+  changeFormat: [formatValue: TextFormatOptions]
+}>()
+
+const textFormatterSelected = ref(TextFormatOptions.Partial)
+watch(textFormatterSelected, (selected) => emit('changeFormat', selected))
+
+const open = ref(false)
+</script>
+
+<template>
+  <DropdownMenu v-model:open="open" class="TextFormattingSelector">
+    <template #button><SvgIcon name="paragraph" title="Text Display Options" /> </template>
+
+    <template #entries>
+      <SvgButton
+        name="paragraph"
+        class="full-format"
+        label="Full whitespace rendering"
+        :title="`Text displayed in monospace font and all whitespace characters displayed as symbols`"
+        @click.stop="() => emit('changeFormat', TextFormatOptions.On)"
+      />
+
+      <SvgButton
+        name="paragraph"
+        class="partial"
+        label="Partial whitespace rendering"
+        :title="`Text displayed in monospace font, only multiple spaces displayed with &#183;`"
+        @click.stop="() => emit('changeFormat', TextFormatOptions.Partial)"
+      />
+      <button
```

Comment:
All of our buttons have tooltips and some consistent styling. I see this one matches the styles by copy-and-paste, but that won't stay up to date if we change the rest of the buttons. This button too could be rendered with one of our button abstractions:
- It could be a `SvgButton`. The icon can be styled using a deep-style selector: `.OffButton :deep(.SvgIcon) { /* strikethrough styles */ }`
- Or it could be done with our most general button, `MenuButton`, which can be applied to anything to provide the button styling and tooltip.

Reply:
I have used Menu Button, but for all entries as this seemed to make it easier to have consistent formatting